### PR TITLE
fix radiasoft/raydata#41: analysis run log modal

### PR DESCRIPTION
**Pull request by rorour**
fixes radiasoft/raydata#41
depends on radiasoft/raydata#52

### CODE REVIEW

**Review by e-carlin — CHANGES_REQUESTED** [review]
I haven't looked at the code yet but noticed two things while clicking around the app.

1. There is what looks to me like an extra link to a run log. Is that supposed to be there?
Screen Shot 2022-12-08 at 10 36 21 AM
2. When I click on the run log button I see
Screen Shot 2022-12-08 at 10 39 27 AM
But looking in the console I see a much larger payload returned in the log which reflects the full run.log on disk.

```
{"run_log":"Input Notebook:  /home/vagrant/src/radiasoft/raydata/run/9fd30e8b-c160-4b73-9d95-6441c27535e3/XPCS_SAXS.ipynb\nOutput Notebook: /home/vagrant/src/radiasoft/raydata/run/9fd30e8b-c160-4b73-9d95-6441c27535e3/XPCS_SAXS.ipynb-out.ipynb\n\nExecuting:   0%|          | 0/170 [00:00<?, ?cell/s]Executing notebook with kernel: py3\n\nExecuting:   1%|          | 1/170 [00:07<21:04,  7.48s/cell]\nExecuting:   1%|          | 2/170 [00:12<16:33,  5.91s/cell]\nExecuting:   2%|\u258f         | 3/170 [00:12<09:04,  3.26s/cell]\nExecuting:   2%|\u258f         | 4/170 [00:12<05:38,  2.04s/cell]\nExecuting:   3%|\u258e         | 5/170 [00:12<03:45,  1.37s/cell]\nExecuting:   4%|\u258e         | 6/170 [00:12<02:35,  1.06cell/s]\nExecuting:   4%|\u258d         | 7/170 [00:12<01:50,  1.48cell/s]\nExecuting:   5%|\u258d         | 8/170 [00:13<01:21,  1.98cell/s]\nExecuting:   5%|\u258c         | 9/170 [00:13<01:02,  2.57cell/s]\nExecuting:   6%|\u258c         | 10/170 [00:13<00:49,  3.26cell/s]\nExecuting:   6%|\u258b         | 11/170 [00:13<00:38,  4.09cell/s]\nExecuting:   7%|\u258b         | 12/170 [00:13<00:42,  3.70cell/s]\nExecuting:   8%|\u258a         | 13/170 [00:13<00:34,  4.55cell/s]\nExecuting:   8%|\u258a         | 14/170 [00:14<00:30,  5.14cell/s]\nExecuting:   9%|\u2589         | 16/170 [00:14<00:23,  6.62cell/s]\nExecuting:  10%|\u2588         | 17/170 [00:14<00:23,  6.40cell/s]\nExecuting:  11%|\u2588         | 18/170 [00:14<00:25,  5.96cell/s]\nExecuting:  11%|\u2588         | 19/170 [00:14<00:24,  6.12cell/s]\nExecuting:  12%|\u2588\u258f        | 20/170 [00:14<00:24,  6.12cell/s]\nExecuting:  12%|\u2588\u258f        | 21/170 [00:15<00:23,  6.36cell/s]\nExecuting:  13%|\u2588\u258e        | 22/170 [00:15<00:20,  7.07cell/s]\nExecuting:  14%|\u2588\u258e        | 23/170 [00:15<00:20,  7.08cell/s]\nExecuting:  14%|\u2588\u258d        | 24/170 [00:15<00:19,  7.58cell/s]\nExecuting:  15%|\u2588\u258d        | 25/170 [00:15<00:18,  7.74cell/s]\nExecuting:  15%|\u2588\u258c        | 26/170 [00:16<00:58,  2.48cell/s]\nExecuting:  16%|\u2588\u258c        | 27/170 [00:16<00:55,  2.56cell/s]\nExecuting:  16%|\u2588\u258b        | 28/170 [00:17<00:44,  3.22cell/s]\nExecuting:  17%|\u2588\u258b        | 29/170 [00:17<00:39,  3.55cell/s]\nExecuting:  18%|\u2588\u258a        | 30/170 [00:17<00:32,  4.27cell/s]\nExecuting:  18%|\u2588\u258a        | 31/170 [00:17<00:31,  4.37cell/s]\nExecuting:  19%|\u2588\u2589        | 32/170 [00:18<00:43,  3.15cell/s]\nExecuting:  19%|\u2588\u2589        | 33/170 [00:18<00:34,  3.93cell/s]\nExecuting:  20%|\u2588\u2588        | 34/170 [00:18<00:31,  4.27cell/s]\nExecuting:  21%|\u2588\u2588        | 35/170 [00:19<01:17,  1.74cell/s]\nExecuting:  21%|\u2588\u2588        | 36/170 [00:19<01:00,  2.22cell/s]\nExecuting:  22%|\u2588\u2588\u258f       | 37/170 [00:20<00:46,  2.87cell/s]\nExecuting:  22%|\u2588\u2588\u258f       | 38/170 [00:20<00:36,  3.58cell/s]\nExecuting:  23%|\u2588\u2588\u258e       | 39/170 [00:20<00:34,  3.84cell/s]\nExecuting:  24%|\u2588\u2588\u258e       | 40/170 [00:20<00:30,  4.31cell/s]\nExecuting:  24%|\u2588\u2588\u258d       | 41/170 [00:20<00:26,  4.84cell/s]\nExecuting:  25%|\u2588\u2588\u258d       | 42/170 [00:20<00:24,  5.25cell/s]\nExecuting:  25%|\u2588\u2588\u258c       | 43/170 [00:21<00:21,  5.93cell/s]\nExecuting:  26%|\u2588\u2588\u258c       | 44/170 [00:21<00:20,  6.23cell/s]\nExecuting:  26%|\u2588\u2588\u258b       | 45/170 [00:22<00:51,  2.43cell/s]\nExecuting:  27%|\u2588\u2588\u258b       | 46/170 [00:22<00:41,  2.97cell/s]\nExecuting:  28%|\u2588\u2588\u258a       | 47/170 [00:22<00:34,  3.60cell/s]\nExecuting:  28%|\u2588\u2588\u258a       | 48/170 [00:22<00:30,  3.99cell/s]\nExecuting:  29%|\u2588\u2588\u2589       | 49/170 [00:31<05:50,  2.90s/cell]\nExecuting:  29%|\u2588\u2588\u2589       | 50/170 [00:32<04:26,  2.22s/cell]\nExecuting:  30%|\u2588\u2588\u2588       | 51/170 [00:32<03:15,  1.65s/cell]\nExecuting:  31%|\u2588\u2588\u2588       | 52/170 [00:33<02:49,  1.44s/cell]\nExecuting:  31%|\u2588\u2588\u2588       | 53/170 [00:33<02:01,  1.04s/cell]\nExecuting:  32%|\u2588\u2588\u2588\u258f      | 54/170 [00:33<01:32,  1.25cell/s]\nExecuting:  32%|\u2588\u2588\u2588\u258f      | 55/170 [00:34<01:09,  1.67cell/s]\nExecuting:  33%|\u2588\u2588\u2588\u258e      | 56/170 [00:34<00:52,  2.18cell/s]\nExecuting:  34%|\u2588\u2588\u2588\u258e      | 57/170 [00:48<08:47,  4.66s/cell]\nExecuting:  34%|\u2588\u2588\u2588\u258d      | 58/170 [00:49<06:28,  3.47s/cell]\nExecuting:  35%|\u2588\u2588\u2588\u258d      | 59/170 [00:49<04:34,  2.47s/cell]\nExecuting:  35%|\u2588\u2588\u2588\u258c      | 60/170 [00:52<04:38,  2.53s/cell]\nExecuting:  36%|\u2588\u2588\u2588\u258c      | 61/170 [00:52<03:17,  1.82s/cell]\nExecuting:  36%|\u2588\u2588\u2588\u258b      | 62/170 [00:52<02:21,  1.31s/cell]\nExecuting:  37%|\u2588\u2588\u2588\u258b      | 63/170 [00:52<01:42,  1.04cell/s]\nExecuting:  38%|\u2588\u2588\u2588\u258a      | 64/170 [00:53<01:44,  1.01cell/s]\nExecuting:  38%|\u2588\u2588\u2588\u258a      | 65/170 [00:53<01:16,  1.36cell/s]\nExecuting:  39%|\u2588\u2588\u2588\u2589      | 66/170 [00:54<00:58,  1.77cell/s]\nExecuting:  39%|\u2588\u2588\u2588\u2589      | 67/170 [00:54<00:44,  2.31cell/s]\nExecuting:  40%|\u2588\u2588\u2588\u2588      | 68/170 [00:54<00:49,  2.04cell/s]\nExecuting:  41%|\u2588\u2588\u2588\u2588      | 69/170 [00:54<00:38,  2.61cell/s]\nExecuting:  41%|\u2588\u2588\u2588\u2588      | 70/170 [00:55<00:32,  3.08cell/s]\nExecuting:  42%|\u2588\u2588\u2588\u2588\u258f     | 71/170 [00:55<00:26,  3.75cell/s]\nExecuting:  42%|\u2588\u2588\u2588\u2588\u258f     | 72/170 [00:55<00:22,  4.40cell/s]\nExecuting:  43%|\u2588\u2588\u2588\u2588\u258e     | 73/170 [00:55<00:19,  4.89cell/s]\nExecuting:  44%|\u2588\u2588\u2588\u2588\u258e     | 74/170 [00:58<01:35,  1.00cell/s]\nExecuting:  44%|\u2588\u2588\u2588\u2588\u258d     | 75/170 [00:58<01:10,  1.35cell/s]\nExecuting:  45%|\u2588\u2588\u2588\u2588\u258d     | 76/170 [00:58<00:53,  1.77cell/s]\nExecuting:  45%|\u2588\u2588\u2588\u2588\u258c     | 77/170 [00:58<00:41,  2.26cell/s]\nExecuting:  46%|\u2588\u2588\u2588\u2588\u258c     | 78/170 [01:03<02:28,  1.61s/cell]\nExecuting:  46%|\u2588\u2588\u2588\u2588\u258b     | 79/170 [01:03<01:47,  1.18s/cell]\nExecuting:  47%|\u2588\u2588\u2588\u2588\u258b     | 80/170 [01:07<03:05,  2.06s/cell]\nExecuting:  48%|\u2588\u2588\u2588\u2588\u258a     | 81/170 [01:07<02:12,  1.49s/cell]\nExecuting:  48%|\u2588\u2588\u2588\u2588\u258a     | 82/170 [01:07<01:35,  1.09s/cell]\nExecuting:  49%|\u2588\u2588\u2588\u2588\u2589     | 83/170 [01:07<01:11,  1.22cell/s]\nExecuting:  49%|\u2588\u2588\u2588\u2588\u2589     | 84/170 [01:08<00:53,  1.62cell/s]\nExecuting:  50%|\u2588\u2588\u2588\u2588\u2588     | 85/170 [01:08<00:40,  2.07cell/s]\nExecuting:  51%|\u2588\u2588\u2588\u2588\u2588     | 86/170 [01:08<00:32,  2.61cell/s]\nExecuting:  51%|\u2588\u2588\u2588\u2588\u2588     | 87/170 [01:08<00:26,  3.18cell/s]\nExecuting:  52%|\u2588\u2588\u2588\u2588\u2588\u258f    | 88/170 [01:08<00:21,  3.73cell/s]\nExecuting:  52%|\u2588\u2588\u2588\u2588\u2588\u258f    | 89/170 [01:08<00:18,  4.30cell/s]\nExecuting:  53%|\u2588\u2588\u2588\u2588\u2588\u258e    | 90/170 [01:09<00:17,  4.66cell/s]\nExecuting:  54%|\u2588\u2588\u2588\u2588\u2588\u258e    | 91/170 [01:09<00:15,  5.01cell/s]\nExecuting:  54%|\u2588\u2588\u2588\u2588\u2588\u258d    | 92/170 [01:09<00:14,  5.40cell/s]\nExecuting:  55%|\u2588\u2588\u2588\u2588\u2588\u258d    | 93/170 [01:09<00:13,  5.59cell/s]\nExecuting:  55%|\u2588\u2588\u2588\u2588\u2588\u258c    | 94/170 [01:09<00:12,  5.95cell/s]\nExecuting:  56%|\u2588\u2588\u2588\u2588\u2588\u258c    | 95/170 [01:09<00:12,  6.04cell/s]\nExecuting:  56%|\u2588\u2588\u2588\u2588\u2588\u258b    | 96/170 [01:09<00:11,  6.23cell/s]\nExecuting:  57%|\u2588\u2588\u2588\u2588\u2588\u258b    | 97/170 [01:10<00:14,  5.03cell/s]\nExecuting:  58%|\u2588\u2588\u2588\u2588\u2588\u258a    | 98/170 [01:10<00:15,  4.80cell/s]\nExecuting:  58%|\u2588\u2588\u2588\u2588\u2588\u258a    | 99/170 [01:10<00:14,  4.96cell/s]\nExecuting:  59%|\u2588\u2588\u2588\u2588\u2588\u2589    | 100/170 [01:10<00:13,  5.33cell/s]\nExecuting:  59%|\u2588\u2588\u2588\u2588\u2588\u2589    | 101/170 [01:11<00:21,  3.27cell/s]\nExecuting:  60%|\u2588\u2588\u2588\u2588\u2588\u2588    | 102/170 [01:11<00:17,  3.88cell/s]\nExecuting:  61%|\u2588\u2588\u2588\u2588\u2588\u2588    | 103/170 [01:11<00:15,  4.30cell/s]Kernel died while waiting for execute reply.\n\nExecuting:  61%|\u2588\u2588\u2588\u2588\u2588\u2588    | 104/170 [01:27<05:27,  4.96s/cell]\nExecuting:  61%|\u2588\u2588\u2588\u2588\u2588\u2588    | 104/170 [01:27<00:55,  1.18cell/s]\nTraceback (most recent call last):\n  File \"/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/nbclient/client.py\", line 1006, in async_execute_cell\n    exec_reply = await self.task_poll_for_reply\nconcurrent.futures._base.CancelledError\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/papermill/cli.py\", line 267, in papermill\n    execution_timeout=execution_timeout,\n  File \"/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/papermill/execute.py\", line 124, in execute_notebook\n    **engine_kwargs\n  File \"/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/papermill/engines.py\", line 49, in execute_notebook_with_engine\n    return self.get_engine(engine_name).execute_notebook(nb, kernel_name, **kwargs)\n  File \"/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/papermill/engines.py\", line 367, in execute_notebook\n    cls.execute_managed_notebook(nb_man, kernel_name, log_output=log_output, **kwargs)\n  File \"/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/papermill/engines.py\", line 436, in execute_managed_notebook\n    return PapermillNotebookClient(nb_man, **final_kwargs).execute()\n  File \"/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/papermill/clientwrap.py\", line 45, in execute\n    self.papermill_execute_cells()\n  File \"/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/papermill/clientwrap.py\", line 72, in papermill_execute_cells\n    self.execute_cell(cell, index)\n  File \"/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/nbclient/util.py\", line 85, in wrapped\n    return just_run(coro(*args, **kwargs))\n  File \"/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/nbclient/util.py\", line 60, in just_run\n    return loop.run_until_complete(coro)\n  File \"/home/vagrant/.pyenv/versions/3.7.2/lib/python3.7/asyncio/base_events.py\", line 584, in run_until_complete\n    return future.result()\n  File \"/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/nbclient/client.py\", line 1010, in async_execute_cell\n    raise DeadKernelError(\"Kernel died\")\nnbclient.exceptions.DeadKernelError: Kernel died\n","state":"completed"}
```

Would it be possible to have the newlines in the payload cause line breaks so the text is better formatted?

**Comment by rorour** [comment]
Hmm, are you sure you aren't looking at an older commit? That's how it looked before it was finished.

File: sirepo/package_data/static/js/sirepo-components.js
Comment (e-carlin) [inline sirepo/package_data/static/js/sirepo-components.js:1884]: It took me a minute to figure out the logic for the view log wrappers. It may be worth setting up a meeting and we can go over it together. But, my basic thought is that this line belongs in the wrappers and not in this directive. If you do that then I think some of the tricky bits of code with setting the scan id and needing a timeout to let a digest cycle happen will go away. Instead we can use bootsrap.show and bootstrap `shown.bs.modal` (if you search you will see other uses of this) to show the modal and then call a callback when it is shown.  Likewise, I think we'll be able to get rid of the hideLink boolean. That should live in the wrappers (or maybe we just change it to "show log" everywhere) but still  I think moving this line into the wrappers will eliminate the need for hiding in the raydata case.

**Review by e-carlin — CHANGES_REQUESTED** [review]
Worked well. I left some comments. The wrapper code is on the right track but I think needs some tweaks. Happy to meet to share my thoughts.

**Review by Benbenbenin0 — COMMENTED** [review]
Reviewed with @e-carlin

File: sirepo/package_data/static/js/raydata.js
Comment (rorour) [reply sirepo/package_data/static/js/raydata.js]: 8451e485b7d89384a1c03dc2174ff822294749ba

**Comment by robnagler** [comment]
@rorour ping

File: sirepo/package_data/static/js/sirepo-components.js
Comment (rorour) [reply sirepo/package_data/static/js/sirepo-components.js]: Is this close to what you had in mind? 27f9e3765b906a03ddb4fa4099ea53ebb2d0272d

File: sirepo/package_data/static/js/raydata.js
Comment (e-carlin) [inline sirepo/package_data/static/js/raydata.js:322]: How about "View Log" instead of "run log"? I think its a bit clearer what will happen (viewing something instead of running something) and we tend to capitalize words like that elsewhere in Sirepo (ex "Save Changes")

File: sirepo/package_data/static/js/sirepo-components.js
Comment (e-carlin) [inline sirepo/package_data/static/js/sirepo-components.js:1878]: How about logIsLoading? "is" implies a boolean state which helps tip off readers that's what this is used for.

File: sirepo/package_data/static/js/sirepo-components.js
Comment (e-carlin) [inline sirepo/package_data/static/js/sirepo-components.js:1884]: I think the id should be "sr-view-log-iframe" since that is the name of the directive.

File: sirepo/package_data/static/js/sirepo-components.js
Comment (e-carlin) [inline sirepo/package_data/static/js/sirepo-components.js:1881]: How about just downloadLog? This directive doesn't need to know that there will be a "wrapper" around it.

**Review by e-carlin — APPROVED** [review]
Couple nits but looks good! Sorry for the ridiculous delay reviewing this. Once you've fixed up the couple notes you can merge this and the companion PR in raydata.

File: sirepo/package_data/static/js/sirepo-components.js
Comment (e-carlin) [reply sirepo/package_data/static/js/sirepo-components.js:1881]: Ah, I didn't see that. Is there a reason for that function? I think we could get rid of it, rename wrapperDownloadLog to downloadLog, and the code should work.

File: sirepo/package_data/static/js/raydata.js
Comment (rorour) [reply sirepo/package_data/static/js/raydata.js]: 4d8ecf81fb4e98997af2c3b6de70315095700cc9

File: sirepo/package_data/static/js/sirepo-components.js
Comment (rorour) [reply sirepo/package_data/static/js/sirepo-components.js]: 4d8ecf81fb4e98997af2c3b6de70315095700cc9

File: sirepo/package_data/static/js/sirepo-components.js
Comment (rorour) [reply sirepo/package_data/static/js/sirepo-components.js]: 4d8ecf81fb4e98997af2c3b6de70315095700cc9

File: sirepo/package_data/static/js/sirepo-components.js
Comment (rorour) [reply sirepo/package_data/static/js/sirepo-components.js]: 4d8ecf81fb4e98997af2c3b6de70315095700cc9

**Comment by rorour** [comment]
Blocked by https://github.com/radiasoft/sirepo/issues/5435

@e-carlin I saw the bug you mentioned the other day. There seemed to be 2 issues. Firstly, there were 2 run log modals with the same id on the queue page. I fixed this by generating a unique id for the modals similar to the analysis modal id.
Secondly, there was a race condition when opening the modal where the scanId binding in the viewLogIframeWrapper didn't update in time. I fixed this by adding a $timeout.
Please take a look at the new changes before I merge.

File: sirepo/package_data/static/js/sirepo-components.js
Comment (rorour) [reply sirepo/package_data/static/js/sirepo-components.js]: d48d4786acf2122b7744c7d0ad57b0e5633445d4

File: sirepo/package_data/static/js/raydata.js
Comment (rorour) [reply sirepo/package_data/static/js/raydata.js]: d48d4786acf2122b7744c7d0ad57b0e5633445d4

**Pull request merged**
